### DISCUSSION TITLE
WASM SIMD not working.

### DISCUSSION THREAD

**Discussion by harutyuntagushyan8** [emscripten-core/emscripten#17597] [General]
I can't figure out when simd instruction are generated. I Have double for loop and build with -msimd128 flag but no performance gain.
My code is same as in this link
https://v8.dev/features/simd#:~:text=For%20example%2C%20consider%20the%20following%20function%20that%20multiplies%20the%20elements%20of%20two%20input%20arrays%20together%20and%20stores%20the%20results%20in%20an%20output%20array.
Steps I have done is same.

**Comment by harutyuntagushyan8** [comment]
When I looked out generated wasm file as watt file there is no v128 type which correspond to 128-bit register.

**Comment by tlively** [comment]
Are you also passing -O2 or -O3? Without high levels of optimization, the compiler will not autovectorize loops like that.

**Reply by harutyuntagushyan8** [reply]
Yes. I'm  passing -O3 and after loop execution I save for example sum of all elements in array and now it's OK. I have simd generated wasm file and really simd is more performant. Without saving result maybe compiler remove intermediate results and measured time mainly is about 0 miliseconds.
Thanks for your response @tlively .
